docroot/modules/custom/va_gov_links/src/EventSubscriber/ContentReleaseBrokenLinksSubscriber.php

```
@@ -0,0 +1,109 @@
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Re-queues a content release when an error condition happens.
+ */
+class ContentReleaseBrokenLinksSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Http Client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+  /**
+   * Logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $logger;
+  /**
+   * Settings Service.
+   *
+   * @var \Drupal\Core\Site\Settings
+   */
+  protected $settings;
+
+  /**
+   * Constructor for ContentReleaseBrokenLinksSubscriber objects.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   Http client.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state service.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger
+   *   Logger logger logger logger.
+   * @param \Drupal\Core\Site\Settings $settings
+   *   Settings.
+   */
+  public function __construct(ClientInterface $http_client, StateInterface $state, LoggerChannelFactoryInterface $logger, Settings $settings) {
+    $this->httpClient = $http_client;
+    $this->state = $state;
+    $this->logger = $logger;
+    $this->settings = $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[ReleaseStateTransitionEvent::NAME] = 'retrieveBrokenLinks';
+    return $events;
+  }
+
+  /**
+   * Retrieve the broken links report when a content release finishes.
+   *
+   * @param \Drupal\va_gov_build_trigger\Event\ReleaseStateTransitionEvent $event
+   *   The release state transition event.
+   */
+  public function retrieveBrokenLinks(ReleaseStateTransitionEvent $event) {
+    if ($this->settings->get('broken_link_report_import_enabled')) {
```

I like that you are basing this on a setting.  It seems like this should be available to a UI  to be able to disable or enable.  In case the data goes wonky and we need to shut it down either to silence the errors or prevent additional broken link processing and notifications (though notifications should have its own shunt).

---

Yeah, a kill switch in State would be nice.  We should probably encourage that for all functionality that differs on BRD, given the time and PITA of the full deployment pipeline and off-cycle deployments.

---

We should have a feature flag functionality to use for something like this. In the absence of that, I can put it into State. I don't think a separate form is warranted for this.